PRICKLE4.py

```diff
@@ -0,0 +1,190 @@
+import os
+from abc import ABC, abstractmethod
+from Bio import SeqIO
+from Bio import SeqUtils
+
+
+class BiologicalSequence(ABC, str):
+
+    def __init__(self, seq):
+        self.seq = seq
+
+    #да, Никита уже сказал, что так лучше не делать, а оставить сдесь только абстрактные методы,
+    # но я зацепилась за идею не повторять проверки, раз суть для всех одна, только алфавиты разные,
+    # а переделать уже не успеваю
+    def check_alphabet(self):
+        if set(self.seq) <= self.alphabet:
+            raise UnexpectedSymbolInSeqError
+
+    # не поняла, не доделала
+    @abstractmethod
+    def to_print_seq(self):
+        return self.seq
+        pass
+

```

Comment:
Абстрактные методы в абстрактном классе определяют некий интерфейс, который будет унаследован дочерними классами. Тут подразумевалось, что в родительском классе `Biological sequence` вы скажите, что дочерние классы должны уметь считать длину, получать индексы и прочее. Например, про длину, это можно записать как 
```suggestion
    @abstractmethod
    def __len__(self):
        pass


```
И потом переопределить нашу заглушку в дочернем, написав что-то осмысленное вместо `pass`
Как-то так)